src/main/java/org/cactoos/text/TextOfIterable.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import org.cactoos.iterable.Mapped;
+
+/**
+ * Text of {@link Iterable}
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 0.49
+ */
+public final class TextOfIterable extends TextEnvelope {
+
+    /**
+     * Ctor.
+     *
+     * @param iterable The iterable to convert to string
+     */
+    public TextOfIterable(final Iterable<?> iterable) {
+        super(
+            new TextOfScalar(
+                () -> new Joined(
+                    ", ",
+                    new Mapped<>(
+                        Object::toString,
+                        iterable
+                    )
+                ).asString()
```

Comment:
@baudoliver7 here the idea is that you can directly pass the Joined object to super without using `TextOfScalar`. that's why there was this todo: to avoid using scalars when not needed.